File: typescript/src/services/file-tasks-api.ts
/** Generated API client. */
import type { ApiResponse } from '../types.js';
import type { DocmostHttpClient } from '../http-client.js';
import type { FileTaskIdDto, PaginationOptions } from '../models/index.js';


export class FileTasksApi {
  constructor(private readonly http: DocmostHttpClient) {}

  async getFileTasks(request: PaginationOptions): Promise<ApiResponse<unknown>> {
    return this.http.post(`file-tasks`, request);
  }

  async getFileTask(request: FileTaskIdDto): Promise<ApiResponse<unknown>> {
    return this.http.post(`file-tasks/info`, request);
  }

}

File: c#/src/Docmost.Sdk/Services/FileTasksApi.cs
using Docmost.Sdk.Http;
using Docmost.Sdk.Models;
using System.Net.Http.Headers;
using System.Net.Http.Json;
using System.Text.Json;

namespace Docmost.Sdk.Services;

public sealed class FileTasksApi(DocmostHttpClient http)
{
    private readonly DocmostHttpClient _http = http;

    public async Task<ApiResponse<JsonElement?>> GetFileTasksAsync(PaginationOptions request, CancellationToken cancellationToken = default)
    {
        return await _http.PostAsync<JsonElement?>("file-tasks", request, cancellationToken).ConfigureAwait(false);
    }

    public async Task<ApiResponse<JsonElement?>> GetFileTaskAsync(FileTaskIdDto request, CancellationToken cancellationToken = default)
    {
        return await _http.PostAsync<JsonElement?>("file-tasks/info", request, cancellationToken).ConfigureAwait(false);
    }

}
